ee/agent/storage/sqlite/keyvalue_store_sqlite.go

```diff
@@ -0,0 +1,213 @@
+package agentsqlite
+
+import (
+	"context"
+	"database/sql"
+	"embed"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/sqlite"
+	"github.com/golang-migrate/migrate/v4/source/iofs"
+	_ "modernc.org/sqlite"
+)
+
+//go:embed migrations/*.sqlite
+var migrations embed.FS
+
+type SqliteStore struct {
+	conn          *sql.DB
+	rootDirectory string
+}
+
+func NewStore(ctx context.Context, rootDirectory string) (*SqliteStore, error) {
+	conn, err := dbConn(ctx, rootDirectory)
+	if err != nil {
+		return nil, fmt.Errorf("opening startup db in %s: %w", rootDirectory, err)
+	}
+
+	s := &SqliteStore{
+		conn:          conn,
+		rootDirectory: rootDirectory,
+	}
+
+	if err := s.migrate(ctx); err != nil {
+		return nil, fmt.Errorf("migrating the database: %w", err)
+	}
+
+	return s, nil
+}
+
+// dbConn returns a connection to the database in the given rootDirectory.
+// It will create a database there if one does not yet exist.
+func dbConn(ctx context.Context, rootDirectory string) (*sql.DB, error) {
+	// Ensure db exists
+	startupDbFilepath := dbLocation(rootDirectory)
+	if _, err := os.Stat(startupDbFilepath); os.IsNotExist(err) {
+		f, err := os.Create(startupDbFilepath)
+		if err != nil {
+			return nil, fmt.Errorf("creating file at %s: %w", startupDbFilepath, err)
+		}
+		f.Close()
+	}
```

Comment:
are we swallowing all non-`IsNotExist` errors

Reply:
Oh yup you're right -- I don't remember why I decided to do this but I don't think it makes sense anymore. I will update!